parsers/parser.go

```diff
@@ -211,6 +187,29 @@ func toStats(parsedValue map[string]string, keys *statKeys, strictMode, queryStr
 	return NewParsedHTTPStat(uri, method, timestr, resTime, bodyBytes, status), nil
 }
 
+func normalizeURL(src *url.URL, queryString, qsIgnoreValues bool) string {
+	if src.RawQuery == "" {
+		return src.String()
+	}
+
+	u := *src // basic clone
+	if !queryString {
+		u.RawQuery = ""
+		return u.String()
+	}
+
+	if qsIgnoreValues {
+		values := u.Query()
+		for q := range values {
+			values.Set(q, "xxx")
+		}
+		u.RawQuery = values.Encode()
+	} else {
+		u.RawQuery = u.Query().Encode() // re-encode to sort queries
```

Comment:
I just found out that Encode can sort query parameters.
https://pkg.go.dev/net/url#Values.Encode
> Encode encodes the values into “URL encoded” form ("bar=baz&foo=quux") sorted by key.
https://cs.opensource.google/go/go/+/refs/tags/go1.16.6:src/net/url/url.go;l=972